perf/src/packet.rs

```diff
@@ -61,7 +65,7 @@ impl Packets {
 pub fn to_packets_chunked<T: Serialize>(xs: &[T], chunks: usize) -> Vec<Packets> {
     let mut out = vec![];
     for x in xs.chunks(chunks) {
-        let mut p = Packets::default();
+        let mut p = Packets::with_capacity(chunks);
         p.packets.resize(x.len(), Packet::default());
```

Comment:
Ah does the `resize` call here not remove the excess bloat?

Reply:
> (based on the Vec docs, looks like it might only truncate length, not the allocated capacity)

That's correct. You need to call `shrink_to_fit` if you wanted that behavior. But, better off avoiding the overallocation to begin with. :)